packages/govern-core/contracts/OptimisticQueue.sol

```diff
@@ -14,115 +14,157 @@ import "./lib/OptimisticQueueStateLib.sol";
         require(_container.payload.submitter == msg.sender, "queue: bad submitter");
 
         containerHash = ERC3000Data.containerHash(_container.payload.hash(), _configHash);
         queue[containerHash].checkAndSetState(
-            OptimisticQueueStateLib.State.None,
-            OptimisticQueueStateLib.State.Scheduled
+            OptimisticQueueStateLib.State.None, // ensure that the state for this container is None
+            OptimisticQueueStateLib.State.Scheduled // and if so perform a state transition to Scheduled
         );
+        // we don't need to save any more state about the container in storage
+        // we just authenticate the hash and assign it a state, since all future
+        // actions regarding the container will need to provide it as a witness
+        // all witnesses are logged from this contract at least once, so the 
+        // trust assumption should be the same as storing all on-chain (move complexity to clients)
+
+        ERC3000Data.Collateral memory collateral = _container.config.scheduleDeposit;
+        collateral.collectFrom(_container.payload.submitter); // pull collateral from submitter (requires previous approval)
 
-        ERC3000Data.Collateral memory collateral = _container
-            .config
-            .scheduleDeposit;
-        collateral.collectFrom(msg.sender);
         // TODO: pay court tx fee
 
+        // emit an event to ensure data availability of all state that cannot be otherwise fetched (see how config isn't emitted since an observer should already have it)
         emit Scheduled(containerHash, _container.payload, collateral);
     }
 
+    /**
+     * @notice Executes an action after its execution delayed has passed and its state hasn't been altered by a challenge or veto
+     * @param _container A ERC3000Data.Container struct holding both the paylaod being scheduled for execution and
+       the current configuration of the system
+     */
     function execute(ERC3000Data.Container memory _container)
         public
         override
-        auth(this.execute.selector)
+        auth(this.execute.selector) // in most instances this will be open for any addr, but leaving configurable for flexibility
         returns (bytes[] memory execResults)
     {
+        // ensure enough time has passed
         require(uint64(block.timestamp) >= _container.payload.executionTime, "queue: wait more");
 
         bytes32 containerHash = _container.hash();
         queue[containerHash].checkAndSetState(
-            OptimisticQueueStateLib.State.Scheduled,
+            OptimisticQueueStateLib.State.Scheduled, // note that we will revert here if the container wasn't previously scheduled
             OptimisticQueueStateLib.State.Executed
         );
 
-        _container.config.scheduleDeposit.releaseTo(_container.payload.submitter);
+        _container.config.scheduleDeposit.releaseTo(_container.payload.submitter); // release collateral to executor
```

Comment:
I wonder why we can call `releaseTo` method here without instantiating `ERC3000Data.Collateral memory collateral = _container.config.scheduleDeposit` like we do for `collectFrom`.

Reply:
I don't understand the question